internal/controller/csiaddons/csiaddonsnode_controller.go

```diff
@@ -271,3 +273,21 @@ func validateCSIAddonsNodeSpec(csiaddonsnode *csiaddonsv1alpha1.CSIAddonsNode) e
 
 	return nil
 }
+
+// parseCapabilities returns a list of capabilities in the format
+// capability.Type
+func parseCapabilities(caps []*identity.Capability) []string {
+	if len(caps) == 0 {
+		return []string{}
+	}
+
+	capabilities := make([]string, 0, len(caps))
+
+	for _, cap := range caps {
+		capStr := strings.ReplaceAll(cap.String(), ":{type:", ".")
+		capStr = strings.ReplaceAll(capStr, "}", "")
+		capabilities = append(capabilities, capStr)
```

Comment:
you already created a right sized slice, no need to re-create a new `capabilities` slice in every loop. You can be a little more efficient like this:

```go
for i, cap := range caps {
    ...
    capabilities[i] = capStr
}
```

Reply:
Added a fixed slice later as an optimization and forgot to update the rest of the code to actually use that optimization. Classic.

Thank you! Fixed :)